json_object.c

```diff
@@ -1340,7 +1433,14 @@ int json_object_equal(struct json_object* jso1, struct json_object* jso2)
 			return (jso1->o.c_double == jso2->o.c_double);
 
 		case json_type_int:
-			return (jso1->o.c_int64 == jso2->o.c_int64);
+			if ((jso1->o.c_int.cint_type == json_object_int_type_int64)
```

Comment:
int64 and uint64 are comparable.  If they have the same actual value, let's consider them equal regardless of any differences in cint_type.

Reply:
agree, done.